src/broadcasting.jl

```diff
@@ -0,0 +1,25 @@
+# See: https://docs.julialang.org/en/v1/manual/interfaces/#man-interfaces-broadcasting-1
+
+function Base.BroadcastStyle(::Type{<:NamedDimsArray})
```

Comment:
```
help?> Base.Broadcast.ArrayStyle
  Broadcast.ArrayStyle{MyArrayType}() is a BroadcastStyle indicating that an object behaves as an array for
  broadcasting. It presents a simple way to construct Broadcast.AbstractArrayStyles for specific AbstractArray
  container types. Broadcast styles created this way lose track of dimensionality; if keeping track is important for
  your type, you should create your own custom Broadcast.AbstractArrayStyle.
```
Should we be making a subtype of `AbstractArrayStyle` instead? Keeping dimensionality seems important. Also it would be nice to keep track of the wrapped array type (or at least its `BroadcastStyle`), as we might want to call a different `similar` based on that (even if we don't for now).

Reply:
This seems to work right now.
So I am for now leaning towards leaving it as it.
If we later find issues with how it currently is, we can change it then.

Reply:
Alright, that's fine.

I think that will probably come up as soon as indexes come into play.